packages/deploy/src/core/CoreContracts.ts

```diff
@@ -915,36 +955,54 @@ export default class EvmCoreDeploy extends AbstractCoreDeploy<config.EvmCoreCont
       connections.map((_, i: number) => this.governanceRouter.domains(i)),
     );
     domainsOnChain.sort();
-    expect(connections.every((v, i) => v === domainsOnChain[i])).to.be.true;
+    checklist.equals(
+      true,
+      connections.every((v, i) => v === domainsOnChain[i]),
+      'All connections in gov router',
+    );
 
     let threw = false;
     try {
       await this.governanceRouter.domains(connections.length);
     } catch (_) {
       threw = true;
     }
-    expect(threw).to.be.true;
+    checklist.equals(true, threw, 'Right amount of connections in gov router');
 
     //  ========= UpgradeBeaconController =========
     // owner
     const beaconOwner = await this.upgradeBeaconController.owner();
-    expect(utils.equalIds(beaconOwner, this.governanceRouter.address)).to.be
-      .true;
-
-    if (remoteDomains.length > 0) {
-      // expect all replicas to have to same implementation and upgradeBeacon
-      const firstReplica = replicas[remoteDomains[0]];
-      const replicaImpl = firstReplica.implementation;
-      const replicaBeacon = firstReplica.beacon;
-      // check every other implementation/beacon matches the first
-      remoteDomains.slice(1).forEach((remoteDomain) => {
-        const replica = replicas[remoteDomain];
-        const implementation = replica.implementation;
-        const beacon = replica.beacon;
-        expect(utils.equalIds(implementation, replicaImpl));
-        expect(utils.equalIds(beacon, replicaBeacon));
-      });
+    checklist.equalIds(
+      beaconOwner,
+      this.governanceRouter.address,
+      'Governance router address',
+    );
+
+    if (remoteDomains.length > 0 && replicas) {
+      if (replicas) {
+        // expect all replicas to have to same implementation and upgradeBeacon
+        const firstReplica = replicas[remoteDomains[0]];
+        const replicaImpl = firstReplica.implementation;
+        const replicaBeacon = firstReplica.beacon;
+        // check every other implementation/beacon matches the first
+        remoteDomains.slice(1).forEach((remoteDomain) => {
+          const replica = replicas[remoteDomain];
+          const implementation = replica.implementation;
+          const beacon = replica.beacon;
+          checklist.equalIds(
+            implementation,
+            replicaImpl,
+            "Replica's implementation",
+          );
+          checklist.equalIds(beacon, replicaBeacon, "Replica's beacon");
+        });
+      }
+    }
+
+    if (checklist.hasErrors()) {
+      throw checklist;
```

Comment:
When it throws, do you see the full `ok` and `errors` arrays?

Reply:
it throws or returns the same object. `ok` and `error` are visible both ways because we always want to know what went good, what went bad